{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "59c5f052-b0ff-4824-a1df-479a60366336",
   "metadata": {},
   "outputs": [],
   "source": [
    "from env import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9020cb0",
   "metadata": {},
   "source": [
    "#### Declare sequence to search against."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "eddb512d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence = \"HSQKRVVVLGSGVIGLSSALILARKGYSVHILARDLPEDVSSQTFASPWAGANWTPFMTLTDGPRQAKWEESTFKKWVELVPTGHAMWLKGTRRFAQNEDGLLGHWYKDITPNYRPLPSSECPPGAIGVTYDTLSVHAPKYCQYLARELQKLGATFERRTVTSLEQAFDGADLVVNATGLGAKSIAGIDDQAAEPIRGQTVLVKSPCKRCTMDSSDPASPAYIIPRPGGEVICGGTYGVGDWDLSVNPETVQRILKHCLRLDPTISSDGTIEGIEVLRHNVGLRPARRGGPRVEAERIVLPLDRTKSPLSLGRGSARAAKEKEVTLVHAYGFSSAGYQQSWGAAEDVAQLVDEAFQRYHGAARE\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64d49f21",
   "metadata": {},
   "source": [
    "#### Run BLAST search."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "699e213d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running BLAST search with sequence of length 364...\n",
      "Emailing issues to cookie2004@gmail.com\n",
      "Fetching 5000 full sequences from NCBI...\n",
      "/home/ubuntu/miniconda3/envs/brew/lib/python3.10/site-packages/Bio/Entrez/__init__.py:734: UserWarning: \n",
      "            Email address is not specified.\n",
      "\n",
      "            To make use of NCBI's E-utilities, NCBI requires you to specify your\n",
      "            email address with each request.  As an example, if your email address\n",
      "            is A.N.Other@example.com, you can specify it as follows:\n",
      "               from Bio import Entrez\n",
      "               Entrez.email = 'A.N.Other@example.com'\n",
      "            In case of excessive usage of the E-utilities, NCBI will attempt to contact\n",
      "            a user at the email address provided before blocking access to the\n",
      "            E-utilities.\n",
      "  warnings.warn(\n",
      "Fetched batch 1/100\n",
      "Fetched batch 2/100\n",
      "Fetched batch 3/100\n",
      "Fetched batch 4/100\n",
      "Fetched batch 5/100\n",
      "Fetched batch 6/100\n",
      "Fetched batch 7/100\n",
      "Fetched batch 8/100\n",
      "Fetched batch 9/100\n",
      "Fetched batch 10/100\n",
      "Fetched batch 11/100\n",
      "Fetched batch 12/100\n",
      "Fetched batch 13/100\n",
      "Fetched batch 14/100\n",
      "Fetched batch 15/100\n",
      "Fetched batch 16/100\n",
      "Fetched batch 17/100\n",
      "Fetched batch 18/100\n",
      "Fetched batch 19/100\n",
      "Fetched batch 20/100\n",
      "Fetched batch 21/100\n",
      "Fetched batch 22/100\n",
      "Fetched batch 23/100\n",
      "Fetched batch 24/100\n",
      "Fetched batch 25/100\n",
      "Fetched batch 26/100\n",
      "Fetched batch 27/100\n",
      "Fetched batch 28/100\n",
      "Fetched batch 29/100\n",
      "Fetched batch 30/100\n",
      "Fetched batch 31/100\n",
      "Fetched batch 32/100\n",
      "Fetched batch 33/100\n",
      "Fetched batch 34/100\n",
      "Fetched batch 35/100\n",
      "Fetched batch 36/100\n",
      "Fetched batch 37/100\n",
      "Fetched batch 38/100\n",
      "Fetched batch 39/100\n",
      "Fetched batch 40/100\n",
      "Fetched batch 41/100\n",
      "Fetched batch 42/100\n",
      "Fetched batch 43/100\n",
      "Fetched batch 44/100\n",
      "Fetched batch 45/100\n",
      "Fetched batch 46/100\n",
      "Fetched batch 47/100\n",
      "Fetched batch 48/100\n",
      "Fetched batch 49/100\n",
      "Fetched batch 50/100\n",
      "Fetched batch 51/100\n",
      "Fetched batch 52/100\n",
      "Fetched batch 53/100\n",
      "Fetched batch 54/100\n",
      "Fetched batch 55/100\n",
      "Fetched batch 56/100\n",
      "Fetched batch 57/100\n",
      "Fetched batch 58/100\n",
      "Fetched batch 59/100\n",
      "Fetched batch 60/100\n",
      "Fetched batch 61/100\n",
      "Fetched batch 62/100\n",
      "Fetched batch 63/100\n",
      "Fetched batch 64/100\n",
      "Fetched batch 65/100\n",
      "Fetched batch 66/100\n",
      "Fetched batch 67/100\n",
      "Fetched batch 68/100\n",
      "Fetched batch 69/100\n",
      "Fetched batch 70/100\n",
      "Fetched batch 71/100\n",
      "Fetched batch 72/100\n",
      "Fetched batch 73/100\n",
      "Fetched batch 74/100\n",
      "Fetched batch 75/100\n",
      "Fetched batch 76/100\n",
      "Fetched batch 77/100\n",
      "Fetched batch 78/100\n",
      "Fetched batch 79/100\n",
      "Fetched batch 80/100\n",
      "Fetched batch 81/100\n",
      "Fetched batch 82/100\n",
      "Fetched batch 83/100\n",
      "Fetched batch 84/100\n",
      "Fetched batch 85/100\n",
      "Fetched batch 86/100\n",
      "Fetched batch 87/100\n",
      "Fetched batch 88/100\n",
      "Fetched batch 89/100\n",
      "Fetched batch 90/100\n",
      "Fetched batch 91/100\n",
      "Fetched batch 92/100\n",
      "Fetched batch 93/100\n",
      "Fetched batch 94/100\n",
      "Fetched batch 95/100\n",
      "Fetched batch 96/100\n",
      "Fetched batch 97/100\n",
      "Fetched batch 98/100\n",
      "Fetched batch 99/100\n",
      "Fetched batch 100/100\n",
      "Saved 4998 sequences to /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000.fasta\n",
      "File 'my_blast.xml' deleted successfully.\n",
      "Successfully retrieved and saved 4998 sequences to /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000.fasta\n"
     ]
    }
   ],
   "source": [
    "!python src/blast.py $sequence -o /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000.fasta -num_aligns 5000"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3966a3e",
   "metadata": {},
   "source": [
    "#### Run FIDO."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "d1dddf83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sequences kept: 4807 Sequences removed: 191\n",
      "Average sequence length: 364\n",
      "\n",
      "\n",
      "Building a new DB, current time: 08/02/2025 15:51:35\n",
      "New DB name:   /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/blastp_data/blast_fasta_db\n",
      "New DB title:  /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_1_filtered.fasta\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 4807 sequences in 0.099391 seconds.\n",
      "\n",
      "\n",
      "Alignment complete. Writing to /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/blastp_filtered.fasta\n",
      "easy-cluster /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_2_blastp_filtered.fasta /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/mmseq_hmmer_data/step_3_temp_mmseq tmp --min-seq-id 0.9 \n",
      "\n",
      "MMseqs Version:                     \t13.45111\n",
      "Substitution matrix                 \tnucl:nucleotide.out,aa:blosum62.out\n",
      "Seed substitution matrix            \tnucl:nucleotide.out,aa:VTML80.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \t2147483647\n",
      "Alphabet size                       \tnucl:5,aa:21\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \t1\n",
      "Pseudo count b                      \t1.5\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t4\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Gap open cost                       \tnucl:5,aa:11\n",
      "Gap extension cost                  \tnucl:2,aa:1\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \ttrue\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \tnucl:0.200,aa:0.000\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_2_blastp_filtered.fasta tmp/13826935907513613715/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[4445] 0s 4ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 6ms\n",
      "Create directory tmp/13826935907513613715/clu_tmp\n",
      "cluster tmp/13826935907513613715/input tmp/13826935907513613715/clu tmp/13826935907513613715/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.9 --remove-tmp-files 1 \n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 1\n",
      "linclust tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --alph-size nucl:5,aa:13 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 4478 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 4.48K 0s 31ms     \n",
      "Sort kmer 0h 0m 0s 5ms\n",
      "Sort by rep. sequence 0h 0m 0s 2ms\n",
      "Time for fill: 0h 0m 0s 1ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 68ms\n",
      "rescorediagonal tmp/13826935907513613715/input tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore1 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 4.48K 0s 6ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "clust tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore1 tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 4.48K 0s 5ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3464 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 4.48K 0s 3ms      \n",
      "Add missing connections==========================================>] 98.68% 4.42K eta 0s       \n",
      "[=================================================================] 100.00% 4.48K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 12ms\n",
      "Total time: 0h 0m 0s 14ms\n",
      "\n",
      "Size of the sequence database: 4478\n",
      "Size of the alignment database: 4478\n",
      "Number of clusters: 2619\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 20ms\n",
      "createsubdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/order_redundancy tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createsubdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/order_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "filterdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter1 tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter2 --filter-file tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/order_redundancy --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 2.62K 0s 9ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 22ms\n",
      "rescorediagonal tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter2 tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore2 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 2.62K 0s 6ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 2ms==================>] 99.89% 2.62K eta 0s       \n",
      "Time for processing: 0h 0m 0s 52ms\n",
      "align tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore2 tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/aln --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 2619 type: Aminoacid\n",
      "Target database size: 2619 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 2.62K 0s 583ms    \n",
      "Time for merging to aln: 0h 0m 0s 1ms\n",
      "5597 alignments calculated\n",
      "2926 sequence pairs passed the thresholds (0.522780 of overall calculated)\n",
      "1.117220 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 598ms\n",
      "clust tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/aln tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 2.62K 0s 4ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 307 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 2.62K 0s 2ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 2.62K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 19ms\n",
      "Total time: 0h 0m 0s 24ms\n",
      "\n",
      "Size of the sequence database: 2619\n",
      "Size of the alignment database: 2619\n",
      "Number of clusters: 2358\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 58ms\n",
      "mergeclusters tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pre_clust tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/clust --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 2.62K 0s 3ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 2.36K 0s 6ms      \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 4.48K 0s 9ms      \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 18ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pre_clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_filter2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/pref_rescore2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/aln -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/linclust/1223697747079935019/clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_redundancy tmp/13826935907513613715/input tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "prefilter tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/pref_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out --seed-sub-mat nucl:nucleotide.out,aa:VTML80.out -s 1 -k 0 --k-score 2147483647 --alph-size nucl:5,aa:21 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca 1 --pcb 1.5 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 2358 type: Aminoacid\n",
      "Estimated memory consumption: 984M\n",
      "Target database size: 2358 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 2.36K 0s 84ms     \n",
      "Index table: Masked residues: 443\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 2.36K 0s 49ms     \n",
      "Index statistics\n",
      "Entries:          420073\n",
      "DB size:          490 MB\n",
      "Avg k-mer size:   0.006564\n",
      "Top 10 k-mers\n",
      "    NYHGYQ\t732\n",
      "    RPRGRV\t610\n",
      "    HMGYEY\t596\n",
      "    HNGGGY\t595\n",
      "    VCNALP\t499\n",
      "    YPRQLV\t482\n",
      "    GLPRGV\t434\n",
      "    HAGGGY\t319\n",
      "    GGIGYQ\t306\n",
      "    PWGNPV\t302\n",
      "Time for index table init: 0h 0m 0s 842ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 2358\n",
      "Target db start 1 to 2358\n",
      "[=================================================================] 100.00% 2.36K 0s 148ms    \n",
      "\n",
      "2.260768 k-mers per position\n",
      "8812 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 2s 375ms\n",
      "align tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/pref_step0 tmp/13826935907513613715/clu_tmp/2503375529228527601/aln_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 2358 type: Aminoacid\n",
      "Target database size: 2358 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 2.36K 6s 254ms    \n",
      "Time for merging to aln_step0: 0h 0m 0s 2ms\n",
      "45202 alignments calculated\n",
      "2768 sequence pairs passed the thresholds (0.061236 of overall calculated)\n",
      "1.173876 hits per query sequence\n",
      "Time for processing: 0h 0m 6s 271ms\n",
      "clust tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/aln_step0 tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 2.36K 0s 5ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 2.36K 0s 2ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 2.36K 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 12ms\n",
      "Total time: 0h 0m 0s 13ms\n",
      "\n",
      "Size of the sequence database: 2358\n",
      "Size of the alignment database: 2358\n",
      "Number of clusters: 2224\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 15ms\n",
      "mergeclusters tmp/13826935907513613715/input tmp/13826935907513613715/clu tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_redundancy tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_step0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 2.36K 0s 2ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 2.22K 0s 4ms      \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 4.48K 0s 8ms      \n",
      "Time for merging to clu: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/pref_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/aln_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_tmp/2503375529228527601/clu_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createtsv tmp/13826935907513613715/input tmp/13826935907513613715/input tmp/13826935907513613715/clu tmp/13826935907513613715/cluster.tsv --threads 4 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 10ms\n",
      "result2repseq tmp/13826935907513613715/input tmp/13826935907513613715/clu tmp/13826935907513613715/clu_rep --db-load-mode 0 --compressed 0 --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 2.22K 0s 7ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 5ms\n",
      "Time for processing: 0h 0m 0s 57ms\n",
      "result2flat tmp/13826935907513613715/input tmp/13826935907513613715/input tmp/13826935907513613715/clu_rep tmp/13826935907513613715/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "createseqfiledb tmp/13826935907513613715/input tmp/13826935907513613715/clu tmp/13826935907513613715/clu_seqs --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 2.22K 0s 7ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 0m 0s 48ms\n",
      "result2flat tmp/13826935907513613715/input tmp/13826935907513613715/input tmp/13826935907513613715/clu_seqs tmp/13826935907513613715/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 23ms\n",
      "rmdb tmp/13826935907513613715/input -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/input_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_seqs -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu_rep -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/13826935907513613715/clu -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Mseq2 clustering complete. Writing data to /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/\n",
      "WARNING: Unsupported old command line option '-force' will be ignored (may change default output stream and format)\n",
      "WARNING: Your old-style command-line options were converted to:  ./clustalo -i /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta --dealign -o /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_4_rep_seq_aligned.fasta --outfmt=fasta -v --force\n",
      "Using 4 threads\n",
      "Read 2225 sequences (type: Protein) from /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta\n",
      "Dealigning already aligned input sequences as requested.\n",
      "Using 123 seeds (chosen with constant stride from length sorted seqs) for mBed (from a total of 2225 sequences)\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 38.65u 0.02s 00:00:38.67 Elapsed: 00:00:10\n",
      "mBed created 39 cluster/s (with a minimum of 1 and a soft maximum of 100 sequences each)\n",
      "Distance calculation within sub-clusters done. CPU time: 11.26u 0.00s 00:00:11.26 Elapsed: 00:00:03\n",
      "Guide-tree computation (mBed) done.\n",
      "Progressive alignment progress done. CPU time: 278.91u 2.31s 00:04:41.22 Elapsed: 00:01:45\n",
      "Alignment written to /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_4_rep_seq_aligned.fasta\n",
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/step_4_rep_seq_aligned.fasta\n",
      "# output HMM file:                  /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/rep_seq_hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     step_4_rep_seq_aligned  2225  2110   746    18.39  0.590 \n",
      "\n",
      "# CPU time: 0.72u 0.01s 00:00:00.73 Elapsed: 00:00:00.75\n",
      "HMM-based alignment of fasta database is complete.\n"
     ]
    }
   ],
   "source": [
    "!python run_fido.py \\\n",
    "            -in /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000.fasta \\\n",
    "            -ref /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/oxda_rhoto.fasta \\\n",
    "            -out /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/results/oxda_rhoto_top_5000/ \\\n",
    "            -add_ref /home/ubuntu/cookstore/EVOLVE/test/oxda_rhoto/oxda_rhoto.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f36e62e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "brew",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
