{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "59c5f052-b0ff-4824-a1df-479a60366336",
   "metadata": {},
   "outputs": [],
   "source": [
    "from env import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "eddb512d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence = \"HSQKRVVVLGSGVIGLSSALILARKGYSVHILARDLPEDVSSQTFASPWAGANWTPFMTLTDGPRQAKWEESTFKKWVELVPTGHAMWLKGTRRFAQNEDGLLGHWYKDITPNYRPLPSSECPPGAIGVTYDTLSVHAPKYCQYLARELQKLGATFERRTVTSLEQAFDGADLVVNATGLGAKSIAGIDDQAAEPIRGQTVLVKSPCKRCTMDSSDPASPAYIIPRPGGEVICGGTYGVGDWDLSVNPETVQRILKHCLRLDPTISSDGTIEGIEVLRHNVGLRPARRGGPRVEAERIVLPLDRTKSPLSLGRGSARAAKEKEVTLVHAYGFSSAGYQQSWGAAEDVAQLVDEAFQRYHGAARE\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "699e213d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running BLAST search with sequence of length 364...\n",
      "Emailing issues to cookie2004@gmail.com\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ecook/FIDO/src/blast.py\", line 133, in <module>\n",
      "    blaster = BlastP(args.input, args.output, hitnum=int(args.number_of_alignments))\n",
      "  File \"/home/ecook/FIDO/src/blast.py\", line 18, in __init__\n",
      "    self.data = self.protein_blast(seq)\n",
      "  File \"/home/ecook/FIDO/src/blast.py\", line 42, in protein_blast\n",
      "    return Blast.qblast(\"blastp\", \"nr\", seq, alignments = self.hitnum, hitlist_size=self.hitnum, expect = 0.0001, descriptions=self.hitnum)\n",
      "  File \"/home/ecook/miniconda3/envs/brew/lib/python3.10/site-packages/Bio/Blast/__init__.py\", line 1236, in qblast\n",
      "    time.sleep(wait)\n",
      "KeyboardInterrupt\n",
      "^C\n"
     ]
    }
   ],
   "source": [
    "!python src/blast.py $sequence -o /home/ecook/FIDO/test/results/oxda_rhoto_top_50.fasta -num_aligns 50"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d1dddf83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/ecook/FIDO/test/results/’: File exists\n",
      "Sequences kept: 46 Sequences removed: 3\n",
      "Average sequence length: 383\n",
      "mkdir: cannot create directory ‘/home/ecook/FIDO/test/results/blastp_data’: File exists\n",
      "\n",
      "\n",
      "Building a new DB, current time: 06/26/2025 08:50:18\n",
      "New DB name:   /home/ecook/FIDO/test/results/blastp_data/blast_fasta_db\n",
      "New DB title:  /home/ecook/FIDO/test/results/step_1_filtered.fasta\n",
      "Sequence type: Protein\n",
      "Deleted existing Protein BLAST database named /home/ecook/FIDO/test/results/blastp_data/blast_fasta_db\n",
      "Keep MBits: T\n",
      "Maximum file size: 3000000000B\n",
      "Adding sequences from FASTA; added 46 sequences in 0.000755072 seconds.\n",
      "\n",
      "\n",
      "Alignment complete. Writing to /home/ecook/FIDO/test/results/blastp_filtered.fasta\n",
      "mkdir: cannot create directory ‘/home/ecook/FIDO/test/results/mmseq_hmmer_data’: File exists\n",
      "easy-cluster /home/ecook/FIDO/test/results/step_2_blastp_filtered.fasta /home/ecook/FIDO/test/results/mmseq_hmmer_data/step_3_temp_mmseq tmp --min-seq-id 0.9 \n",
      "\n",
      "MMseqs Version:                     \t16.747c6\n",
      "Substitution matrix                 \taa:blosum62.out,nucl:nucleotide.out\n",
      "Seed substitution matrix            \taa:VTML80.out,nucl:nucleotide.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "Target search mode                  \t0\n",
      "k-score                             \tseq:2147483647,prof:2147483647\n",
      "Alphabet size                       \taa:21,nucl:5\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask residues probability           \t0.9\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Selected taxa                       \t\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \tsubstitution:1.100,context:1.400\n",
      "Pseudo count b                      \tsubstitution:4.100,context:5.800\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t2\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Correlation score weight            \t0\n",
      "Gap open cost                       \taa:11,nucl:5\n",
      "Gap extension cost                  \taa:1,nucl:2\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Weight file name                    \t\n",
      "Cluster Weight threshold            \t0.9\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \ttrue\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \taa:0.000,nucl:0.200\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb /home/ecook/FIDO/test/results/step_2_blastp_filtered.fasta tmp/16705791873536101335/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Create directory tmp/16705791873536101335/clu_tmp\n",
      "cluster tmp/16705791873536101335/input tmp/16705791873536101335/clu tmp/16705791873536101335/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.9 --remove-tmp-files 1 \n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 1\n",
      "linclust tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --alph-size aa:13,nucl:5 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "kmermatcher tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --alph-size aa:13,nucl:5 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale aa:0.000,nucl:0.200 --adjust-kmer-len 0 --mask 0 --mask-prob 0.9 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Database size: 46 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Sort kmer 0h 0m 0s 0ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 3ms\n",
      "rescorediagonal tmp/16705791873536101335/input tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore1 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "clust tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore1 tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 15 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 0ms\n",
      "Total time: 0h 0m 0s 1ms\n",
      "\n",
      "Size of the sequence database: 46\n",
      "Size of the alignment database: 46\n",
      "Number of clusters: 36\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createsubdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/order_redundancy tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/order_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "filterdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter1 tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter2 --filter-file tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/order_redundancy --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "rescorediagonal tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter2 tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore2 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "align tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore2 tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/aln --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 36 type: Aminoacid\n",
      "Target database size: 36 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 36 0s 1ms      \n",
      "Time for merging to aln: 0h 0m 0s 0ms\n",
      "45 alignments calculated\n",
      "36 sequence pairs passed the thresholds (0.800000 of overall calculated)\n",
      "1.000000 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 3ms\n",
      "clust tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/aln tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 0ms\n",
      "Total time: 0h 0m 0s 0ms\n",
      "\n",
      "Size of the sequence database: 36\n",
      "Size of the alignment database: 36\n",
      "Number of clusters: 36\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "mergeclusters tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pre_clust tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/clust --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pre_clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_filter2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/pref_rescore2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/aln -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/linclust/12523135234759416846/clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_redundancy tmp/16705791873536101335/input tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "prefilter tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/pref_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' --seed-sub-mat 'aa:VTML80.out,nucl:nucleotide.out' -s 1 -k 0 --target-search-mode 0 --k-score seq:2147483647,prof:2147483647 --alph-size aa:21,nucl:5 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --comp-bias-corr-scale 1 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-prob 0.9 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 36 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 36 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 36 0s 1ms      \n",
      "Index table: Masked residues: 0\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 36 0s 1ms      \n",
      "Index statistics\n",
      "Entries:          6051\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000095\n",
      "Top 10 k-mers\n",
      "    PWGNPF\t32\n",
      "    PRGECG\t30\n",
      "    GLPRGP\t30\n",
      "    YIPPEV\t26\n",
      "    FAPANW\t26\n",
      "    HAGGGY\t20\n",
      "    LVSCCT\t16\n",
      "    ASWGWT\t16\n",
      "    HCRDIS\t14\n",
      "    PGEIGT\t13\n",
      "Time for index table init: 0h 0m 0s 493ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 36\n",
      "Target db start 1 to 36\n",
      "[=================================================================] 100.00% 36 0s 1ms      \n",
      "\n",
      "2.088112 k-mers per position\n",
      "957 DB matches per sequence\n",
      "0 overflows\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 267ms\n",
      "align tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/pref_step0 tmp/16705791873536101335/clu_tmp/12087031216161626179/aln_step0 --sub-mat 'aa:blosum62.out,nucl:nucleotide.out' -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --comp-bias-corr-scale 1 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca substitution:1.100,context:1.400 --pcb substitution:4.100,context:5.800 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --corr-score-weight 0 --gap-open aa:11,nucl:5 --gap-extend aa:1,nucl:2 --zdrop 40 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 36 type: Aminoacid\n",
      "Target database size: 36 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 36 0s 73ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 0ms\n",
      "691 alignments calculated\n",
      "40 sequence pairs passed the thresholds (0.057887 of overall calculated)\n",
      "1.111111 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 77ms\n",
      "clust tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/aln_step0 tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 2 --compressed 0 -v 3 --cluster-weight-threshold 0.9 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 0ms\n",
      "Total time: 0h 0m 0s 0ms\n",
      "\n",
      "Size of the sequence database: 36\n",
      "Size of the alignment database: 36\n",
      "Number of clusters: 34\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "mergeclusters tmp/16705791873536101335/input tmp/16705791873536101335/clu tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_redundancy tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_step0 --threads 2 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 36 0s 0ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 34 0s 0ms      \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 46 0s 0ms      \n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/pref_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/aln_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_tmp/12087031216161626179/clu_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createtsv tmp/16705791873536101335/input tmp/16705791873536101335/input tmp/16705791873536101335/clu tmp/16705791873536101335/cluster.tsv --threads 2 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "result2repseq tmp/16705791873536101335/input tmp/16705791873536101335/clu tmp/16705791873536101335/clu_rep --db-load-mode 0 --compressed 0 --threads 2 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 34 0s 0ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "result2flat tmp/16705791873536101335/input tmp/16705791873536101335/input tmp/16705791873536101335/clu_rep tmp/16705791873536101335/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createseqfiledb tmp/16705791873536101335/input tmp/16705791873536101335/clu tmp/16705791873536101335/clu_seqs --threads 2 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 34 0s 0ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "result2flat tmp/16705791873536101335/input tmp/16705791873536101335/input tmp/16705791873536101335/clu_seqs tmp/16705791873536101335/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/input -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/input_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_seqs -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu_rep -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/16705791873536101335/clu -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Mseq2 clustering complete. Writing data to /home/ecook/FIDO/test/results/\n",
      "WARNING: Unsupported old command line option '-force' will be ignored (may change default output stream and format)\n",
      "WARNING: Your old-style command-line options were converted to:  /home/ecook/programs/clustalo -i /home/ecook/FIDO/test/results/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta --dealign -o /home/ecook/FIDO/test/results/step_4_rep_seq_aligned.fasta --outfmt=fasta -v --force\n",
      "Using 2 threads\n",
      "Read 35 sequences (type: Protein) from /home/ecook/FIDO/test/results/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta\n",
      "not more sequences (35) than cluster-size (100), turn off mBed\n",
      "Dealigning already aligned input sequences as requested.\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 0.05u 0.00s 00:00:00.05 Elapsed: 00:00:00\n",
      "Guide-tree computation done.\n",
      "Progressive alignment progress done. CPU time: 0.93u 0.06s 00:00:00.99 Elapsed: 00:00:01\n",
      "Alignment written to /home/ecook/FIDO/test/results/step_4_rep_seq_aligned.fasta\n",
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             /home/ecook/FIDO/test/results/step_4_rep_seq_aligned.fasta\n",
      "# output HMM file:                  /home/ecook/FIDO/test/results/rep_seq_hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     step_4_rep_seq_aligned    35   797   386     0.82  0.589 \n",
      "\n",
      "# CPU time: 0.07u 0.00s 00:00:00.07 Elapsed: 00:00:00.07\n",
      "HMM-based alignment of fasta database is complete.\n"
     ]
    }
   ],
   "source": [
    "!python /home/ecook/FIDO/run_fido.py \\\n",
    "            -in /home/ecook/FIDO/test/results/oxda_rhoto_top_50.fasta \\\n",
    "            -ref /home/ecook/FIDO/test/oxda_rhoto.fasta \\\n",
    "            -out /home/ecook/FIDO/test/results/ \\\n",
    "            -add_ref /home/ecook/FIDO/test/oxda_rhoto.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f36e62e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "brew",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
