{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "59c5f052-b0ff-4824-a1df-479a60366336",
   "metadata": {},
   "outputs": [],
   "source": [
    "from env import *"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a9020cb0",
   "metadata": {},
   "source": [
    "#### Declare sequence to search against."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "eddb512d",
   "metadata": {},
   "outputs": [],
   "source": [
    "sequence = \"DPKFEAAYDFPGSGSSSELPLKKGDIVFISRDEPSGWSLAKLLDGSKEGWVPTAYMTPYK\""
   ]
  },
  {
   "cell_type": "markdown",
   "id": "64d49f21",
   "metadata": {},
   "source": [
    "#### Run BLAST search."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "699e213d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running BLAST search with sequence of length 60...\n",
      "Emailing issues to cookie2004@gmail.com\n",
      "Fetching 964 full sequences from NCBI...\n",
      "/home/ubuntu/miniconda3/envs/brew/lib/python3.10/site-packages/Bio/Entrez/__init__.py:734: UserWarning: \n",
      "            Email address is not specified.\n",
      "\n",
      "            To make use of NCBI's E-utilities, NCBI requires you to specify your\n",
      "            email address with each request.  As an example, if your email address\n",
      "            is A.N.Other@example.com, you can specify it as follows:\n",
      "               from Bio import Entrez\n",
      "               Entrez.email = 'A.N.Other@example.com'\n",
      "            In case of excessive usage of the E-utilities, NCBI will attempt to contact\n",
      "            a user at the email address provided before blocking access to the\n",
      "            E-utilities.\n",
      "  warnings.warn(\n",
      "Fetched batch 1/20\n",
      "Fetched batch 2/20\n",
      "Fetched batch 3/20\n",
      "Fetched batch 4/20\n",
      "Fetched batch 5/20\n",
      "Fetched batch 6/20\n",
      "Fetched batch 7/20\n",
      "Fetched batch 8/20\n",
      "Fetched batch 9/20\n",
      "Fetched batch 10/20\n",
      "Fetched batch 11/20\n",
      "Fetched batch 12/20\n",
      "Fetched batch 13/20\n",
      "Fetched batch 14/20\n",
      "Fetched batch 15/20\n",
      "Fetched batch 16/20\n",
      "Fetched batch 17/20\n",
      "Fetched batch 18/20\n",
      "Fetched batch 19/20\n",
      "Fetched batch 20/20\n",
      "Saved 961 sequences to /home/ubuntu/cookstore/250603_protein_gym/res/MYO3_YEAST_Tsuboyama_2023_2BTT/top5000.fasta\n",
      "File 'my_blast.xml' deleted successfully.\n",
      "Successfully retrieved and saved 961 sequences to /home/ubuntu/cookstore/250603_protein_gym/res/MYO3_YEAST_Tsuboyama_2023_2BTT/top5000.fasta\n"
     ]
    }
   ],
   "source": [
    "!python src/blast.py $sequence -o /home/ubuntu/cookstore/250603_protein_gym/res/MYO3_YEAST_Tsuboyama_2023_2BTT/top5000.fasta -num_aligns 5000"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e3966a3e",
   "metadata": {},
   "source": [
    "#### Run FIDO."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf9620c1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/’: File exists\n",
      "Sequences kept: 950 Sequences removed: 1\n",
      "Average sequence length: 75\n",
      "\n",
      "\n",
      "Building a new DB, current time: 09/06/2025 19:24:35\n",
      "New DB name:   /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/blastp_data/blast_fasta_db\n",
      "New DB title:  /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_1_filtered.fasta\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 950 sequences in 0.014116 seconds.\n",
      "\n",
      "\n",
      "Alignment complete. Writing to /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/blastp_filtered.fasta\n",
      "easy-cluster /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_2_blastp_filtered.fasta /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/mmseq_hmmer_data/step_3_temp_mmseq tmp --min-seq-id 0.9 \n",
      "\n",
      "MMseqs Version:                     \t13.45111\n",
      "Substitution matrix                 \tnucl:nucleotide.out,aa:blosum62.out\n",
      "Seed substitution matrix            \tnucl:nucleotide.out,aa:VTML80.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \t2147483647\n",
      "Alphabet size                       \tnucl:5,aa:21\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \t1\n",
      "Pseudo count b                      \t1.5\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t4\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Gap open cost                       \tnucl:5,aa:11\n",
      "Gap extension cost                  \tnucl:2,aa:1\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \ttrue\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \tnucl:0.200,aa:0.000\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_2_blastp_filtered.fasta tmp/18318509349619548911/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[910] 0s 0ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "Create directory tmp/18318509349619548911/clu_tmp\n",
      "cluster tmp/18318509349619548911/input tmp/18318509349619548911/clu tmp/18318509349619548911/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.9 --remove-tmp-files 1 \n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 1\n",
      "linclust tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --alph-size nucl:5,aa:13 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 951 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 951 0s 12ms     \n",
      "Sort kmer 0h 0m 0s 15ms\n",
      "Sort by rep. sequence 0h 0m 0s 25ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 83ms\n",
      "rescorediagonal tmp/18318509349619548911/input tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore1 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 951 0s 5ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 1ms=======>           ] 81.68% 777 eta 0s       \n",
      "Time for processing: 0h 0m 0s 41ms\n",
      "clust tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore1 tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 951 0s 4ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 1451 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 951 0s 7ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 951 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 40ms\n",
      "Total time: 0h 0m 0s 58ms\n",
      "\n",
      "Size of the sequence database: 951\n",
      "Size of the alignment database: 951\n",
      "Number of clusters: 370\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 90ms\n",
      "createsubdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/order_redundancy tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createsubdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/order_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "filterdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter1 tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter2 --filter-file tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/order_redundancy --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 370 0s 4ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 12ms\n",
      "rescorediagonal tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter2 tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore2 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 370 0s 1ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 1ms                   ] 37.67% 140 eta 0s       \n",
      "Time for processing: 0h 0m 0s 19ms\n",
      "align tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore2 tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/aln --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 370 type: Aminoacid\n",
      "Target database size: 370 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 370 0s 13ms     \n",
      "Time for merging to aln: 0h 0m 0s 2ms\n",
      "955 alignments calculated\n",
      "379 sequence pairs passed the thresholds (0.396859 of overall calculated)\n",
      "1.024324 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 49ms\n",
      "clust tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/aln tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 370 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 9 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 370 0s 6ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 370 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 69ms\n",
      "\n",
      "Size of the sequence database: 370\n",
      "Size of the alignment database: 370\n",
      "Number of clusters: 362\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 99ms\n",
      "mergeclusters tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pre_clust tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/clust --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 370 0s 8ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 362 0s 42ms     ======================>                ] 73.96% 268 eta 0s       \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 951 0s 58ms     \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 104ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pre_clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_filter2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/pref_rescore2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/aln -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/linclust/17652898616039069645/clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_redundancy tmp/18318509349619548911/input tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/pref_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out --seed-sub-mat nucl:nucleotide.out,aa:VTML80.out -s 1 -k 0 --k-score 2147483647 --alph-size nucl:5,aa:21 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca 1 --pcb 1.5 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 362 type: Aminoacid\n",
      "Estimated memory consumption: 978M\n",
      "Target database size: 362 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 362 0s 7ms      \n",
      "Index table: Masked residues: 44\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 362 0s 4ms      \n",
      "Index statistics\n",
      "Entries:          6793\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000106\n",
      "Top 10 k-mers\n",
      "    EKEQRQ\t80\n",
      "    YLNRFK\t58\n",
      "    TPKKQK\t45\n",
      "    VDENTP\t42\n",
      "    KNRSNQ\t36\n",
      "    DPGDPQ\t35\n",
      "    LREIFR\t32\n",
      "    NDTEKQ\t20\n",
      "    RQYKRS\t20\n",
      "    SFNFMT\t17\n",
      "Time for index table init: 0h 0m 0s 673ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 362\n",
      "Target db start 1 to 362\n",
      "[=================================================================] 100.00% 362 0s 4ms      \n",
      "\n",
      "0.460971 k-mers per position\n",
      "184 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "18 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 1s 983ms\n",
      "align tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/pref_step0 tmp/18318509349619548911/clu_tmp/8352628282637436067/aln_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 362 type: Aminoacid\n",
      "Target database size: 362 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 362 0s 60ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 4ms\n",
      "5965 alignments calculated\n",
      "426 sequence pairs passed the thresholds (0.071417 of overall calculated)\n",
      "1.176796 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 95ms\n",
      "clust tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/aln_step0 tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 362 0s 7ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 0 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 362 0s 4ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 362 0s 1ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 45ms\n",
      "Total time: 0h 0m 0s 55ms\n",
      "\n",
      "Size of the sequence database: 362\n",
      "Size of the alignment database: 362\n",
      "Number of clusters: 344\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 91ms\n",
      "mergeclusters tmp/18318509349619548911/input tmp/18318509349619548911/clu tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_redundancy tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_step0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 362 0s 8ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 344 0s 9ms      \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 951 0s 11ms     \n",
      "Time for merging to clu: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 41ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/pref_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/aln_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_tmp/8352628282637436067/clu_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createtsv tmp/18318509349619548911/input tmp/18318509349619548911/input tmp/18318509349619548911/clu tmp/18318509349619548911/cluster.tsv --threads 4 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "result2repseq tmp/18318509349619548911/input tmp/18318509349619548911/clu tmp/18318509349619548911/clu_rep --db-load-mode 0 --compressed 0 --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 344 0s 3ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 40ms\n",
      "result2flat tmp/18318509349619548911/input tmp/18318509349619548911/input tmp/18318509349619548911/clu_rep tmp/18318509349619548911/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createseqfiledb tmp/18318509349619548911/input tmp/18318509349619548911/clu tmp/18318509349619548911/clu_seqs --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 344 0s 3ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 13ms\n",
      "result2flat tmp/18318509349619548911/input tmp/18318509349619548911/input tmp/18318509349619548911/clu_seqs tmp/18318509349619548911/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "rmdb tmp/18318509349619548911/input -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/input_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_seqs -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu_rep -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/18318509349619548911/clu -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Mseq2 clustering complete. Writing data to /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/\n",
      "WARNING: Unsupported old command line option '-force' will be ignored (may change default output stream and format)\n",
      "WARNING: Your old-style command-line options were converted to:  ./clustalo -i /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta --dealign -o /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_4_rep_seq_aligned.fasta --outfmt=fasta -v --force\n",
      "Using 4 threads\n",
      "Read 345 sequences (type: Protein) from /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta\n",
      "Dealigning already aligned input sequences as requested.\n",
      "Using 71 seeds (chosen with constant stride from length sorted seqs) for mBed (from a total of 345 sequences)\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 0.61u 0.00s 00:00:00.61 Elapsed: 00:00:01\n",
      "mBed created 5 cluster/s (with a minimum of 1 and a soft maximum of 100 sequences each)\n",
      "Distance calculation within sub-clusters done. CPU time: 0.42u 0.00s 00:00:00.42 Elapsed: 00:00:00\n",
      "Guide-tree computation (mBed) done.\n",
      "Progressive alignment progress done. CPU time: 3.69u 0.04s 00:00:03.73 Elapsed: 00:00:01\n",
      "Alignment written to /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_4_rep_seq_aligned.fasta\n",
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/step_4_rep_seq_aligned.fasta\n",
      "# output HMM file:                  /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/rep_seq_hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     step_4_rep_seq_aligned   345   355   317     1.29  0.590 \n",
      "\n",
      "# CPU time: 0.19u 0.00s 00:00:00.19 Elapsed: 00:00:00.19\n",
      "HMM-based alignment of fasta database is complete.\n"
     ]
    }
   ],
   "source": [
    "!python run_fido.py \\\n",
    "            -in /home/ubuntu/cookstore/250603_protein_gym/res/MYO3_YEAST_Tsuboyama_2023_2BTT/hhblits_reducedQT_4285892.fasta \\\n",
    "            -ref /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/ynzc_bacsu.fasta \\\n",
    "            -out /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/top5000/ \\\n",
    "            -add_ref /home/ubuntu/cookstore/250603_protein_gym/res/YNZC_BACSU_Tsuboyama_2023_2JVD/ynzc_bacsu.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "d1dddf83",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘/home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/’: File exists\n",
      "Sequences kept: 386 Sequences removed: 35\n",
      "Average sequence length: 148\n",
      "\n",
      "\n",
      "Building a new DB, current time: 08/05/2025 14:47:43\n",
      "New DB name:   /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/blastp_data/blast_fasta_db\n",
      "New DB title:  /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_1_filtered.fasta\n",
      "Sequence type: Protein\n",
      "Keep MBits: T\n",
      "Maximum file size: 1000000000B\n",
      "Adding sequences from FASTA; added 386 sequences in 0.016093 seconds.\n",
      "\n",
      "\n",
      "Alignment complete. Writing to /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/blastp_filtered.fasta\n",
      "easy-cluster /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_2_blastp_filtered.fasta /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/mmseq_hmmer_data/step_3_temp_mmseq tmp --min-seq-id 0.9 \n",
      "\n",
      "MMseqs Version:                     \t13.45111\n",
      "Substitution matrix                 \tnucl:nucleotide.out,aa:blosum62.out\n",
      "Seed substitution matrix            \tnucl:nucleotide.out,aa:VTML80.out\n",
      "Sensitivity                         \t4\n",
      "k-mer length                        \t0\n",
      "k-score                             \t2147483647\n",
      "Alphabet size                       \tnucl:5,aa:21\n",
      "Max sequence length                 \t65535\n",
      "Max results per query               \t20\n",
      "Split database                      \t0\n",
      "Split mode                          \t2\n",
      "Split memory limit                  \t0\n",
      "Coverage threshold                  \t0.8\n",
      "Coverage mode                       \t0\n",
      "Compositional bias                  \t1\n",
      "Diagonal scoring                    \ttrue\n",
      "Exact k-mer matching                \t0\n",
      "Mask residues                       \t1\n",
      "Mask lower case residues            \t0\n",
      "Minimum diagonal score              \t15\n",
      "Include identical seq. id.          \tfalse\n",
      "Spaced k-mers                       \t1\n",
      "Preload mode                        \t0\n",
      "Pseudo count a                      \t1\n",
      "Pseudo count b                      \t1.5\n",
      "Spaced k-mer pattern                \t\n",
      "Local temporary path                \t\n",
      "Threads                             \t4\n",
      "Compressed                          \t0\n",
      "Verbosity                           \t3\n",
      "Add backtrace                       \tfalse\n",
      "Alignment mode                      \t3\n",
      "Alignment mode                      \t0\n",
      "Allow wrapped scoring               \tfalse\n",
      "E-value threshold                   \t0.001\n",
      "Seq. id. threshold                  \t0.9\n",
      "Min alignment length                \t0\n",
      "Seq. id. mode                       \t0\n",
      "Alternative alignments              \t0\n",
      "Max reject                          \t2147483647\n",
      "Max accept                          \t2147483647\n",
      "Score bias                          \t0\n",
      "Realign hits                        \tfalse\n",
      "Realign score bias                  \t-0.2\n",
      "Realign max seqs                    \t2147483647\n",
      "Gap open cost                       \tnucl:5,aa:11\n",
      "Gap extension cost                  \tnucl:2,aa:1\n",
      "Zdrop                               \t40\n",
      "Rescore mode                        \t0\n",
      "Remove hits by seq. id. and coverage\tfalse\n",
      "Sort results                        \t0\n",
      "Cluster mode                        \t0\n",
      "Max connected component depth       \t1000\n",
      "Similarity type                     \t2\n",
      "Single step clustering              \tfalse\n",
      "Cascaded clustering steps           \t3\n",
      "Cluster reassign                    \tfalse\n",
      "Remove temporary files              \ttrue\n",
      "Force restart with latest tmp       \tfalse\n",
      "MPI runner                          \t\n",
      "k-mers per sequence                 \t21\n",
      "Scale k-mers per sequence           \tnucl:0.200,aa:0.000\n",
      "Adjust k-mer length                 \tfalse\n",
      "Shift hash                          \t67\n",
      "Include only extendable             \tfalse\n",
      "Skip repeating k-mers               \tfalse\n",
      "Database type                       \t0\n",
      "Shuffle input database              \ttrue\n",
      "Createdb mode                       \t1\n",
      "Write lookup file                   \t0\n",
      "Offset of numeric ids               \t0\n",
      "\n",
      "createdb /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_2_blastp_filtered.fasta tmp/5039645332348957457/input --dbtype 0 --shuffle 1 --createdb-mode 1 --write-lookup 0 --id-offset 0 --compressed 0 -v 3 \n",
      "\n",
      "\u001b[33mShuffle database cannot be combined with --createdb-mode 0\n",
      "\u001b[39m\u001b[33mWe recompute with --shuffle 0\n",
      "\u001b[39mConverting sequences\n",
      "[304] 0s 0ms\n",
      "Time for merging to input_h: 0h 0m 0s 0ms\n",
      "Time for merging to input: 0h 0m 0s 0ms\n",
      "Database type: Aminoacid\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "Create directory tmp/5039645332348957457/clu_tmp\n",
      "cluster tmp/5039645332348957457/input tmp/5039645332348957457/clu tmp/5039645332348957457/clu_tmp --max-seqs 20 -c 0.8 --spaced-kmer-mode 1 --alignment-mode 3 -e 0.001 --min-seq-id 0.9 --remove-tmp-files 1 \n",
      "\n",
      "Set cluster sensitivity to -s 1.000000\n",
      "Set cluster mode SET COVER\n",
      "Set cluster iterations to 1\n",
      "linclust tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --alph-size nucl:5,aa:13 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 -k 0 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --rescore-mode 0 --filter-hits 0 --sort-results 0 --remove-tmp-files 1 --force-reuse 0 \n",
      "\n",
      "kmermatcher tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "kmermatcher tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref --sub-mat nucl:nucleotide.out,aa:blosum62.out --alph-size nucl:5,aa:13 --min-seq-id 0.9 --kmer-per-seq 21 --spaced-kmer-mode 1 --kmer-per-seq-scale nucl:0.200,aa:0.000 --adjust-kmer-len 0 --mask 0 --mask-lower-case 0 --cov-mode 0 -k 0 -c 0.8 --max-seq-len 65535 --hash-shift 67 --split-memory-limit 0 --include-only-extendable 0 --ignore-multi-kmer 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Database size: 387 type: Aminoacid\n",
      "Reduced amino acid alphabet: (A S T) (C) (D B N) (E Q Z) (F Y) (G) (H) (I V) (K R) (L J M) (P) (W) (X) \n",
      "\n",
      "Generate k-mers list for 1 split\n",
      "[=================================================================] 100.00% 387 0s 11ms     \n",
      "Sort kmer 0h 0m 0s 29ms\n",
      "Sort by rep. sequence 0h 0m 0s 0ms\n",
      "Time for fill: 0h 0m 0s 0ms\n",
      "Time for merging to pref: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 92ms\n",
      "rescorediagonal tmp/5039645332348957457/input tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore1 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 0 --wrapped-scoring 0 --filter-hits 0 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 387 0s 5ms      \n",
      "Time for merging to pref_rescore1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 47ms\n",
      "clust tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore1 tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pre_clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 387 0s 6ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 943 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 387 0s 6ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 387 0s 1ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 40ms\n",
      "Total time: 0h 0m 0s 47ms\n",
      "\n",
      "Size of the sequence database: 387\n",
      "Size of the alignment database: 387\n",
      "Number of clusters: 70\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to pre_clust: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 70ms\n",
      "createsubdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/order_redundancy tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "createsubdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/order_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter1 -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to pref_filter1: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "filterdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter1 tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter2 --filter-file tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/order_redundancy --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Filtering using file(s)\n",
      "[=================================================================] 100.00% 70 0s 3ms      \n",
      "Time for merging to pref_filter2: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 8ms\n",
      "rescorediagonal tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter2 tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore2 --sub-mat nucl:nucleotide.out,aa:blosum62.out --rescore-mode 1 --wrapped-scoring 0 --filter-hits 1 -e 0.001 -c 0.8 -a 0 --cov-mode 0 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --add-self-matches 0 --sort-results 0 --db-load-mode 0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 70 0s 0ms      \n",
      "Time for merging to pref_rescore2: 0h 0m 0s 3ms===========>       ] 88.41% 62 eta 0s       \n",
      "Time for processing: 0h 0m 0s 51ms\n",
      "align tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore2 tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/aln --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 70 type: Aminoacid\n",
      "Target database size: 70 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 70 0s 14ms     \n",
      "Time for merging to aln: 0h 0m 0s 0ms\n",
      "144 alignments calculated\n",
      "73 sequence pairs passed the thresholds (0.506944 of overall calculated)\n",
      "1.042857 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 58ms\n",
      "clust tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/aln tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/clust --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 70 0s 0ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 3 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 70 0s 3ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 70 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 35ms\n",
      "Total time: 0h 0m 0s 52ms\n",
      "\n",
      "Size of the sequence database: 70\n",
      "Size of the alignment database: 70\n",
      "Number of clusters: 67\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clust: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 97ms\n",
      "mergeclusters tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pre_clust tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/clust --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 70 0s 5ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 67 0s 29ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 387 0s 45ms     \n",
      "Time for merging to clu_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 86ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore1 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pre_clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_filter2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/pref_rescore2 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/aln -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/linclust/7046971417344810174/clust -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createsubdb tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_redundancy tmp/5039645332348957457/input tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy -v 3 --subdb-mode 1 \n",
      "\n",
      "Time for merging to input_step_redundancy: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "prefilter tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/pref_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out --seed-sub-mat nucl:nucleotide.out,aa:VTML80.out -s 1 -k 0 --k-score 2147483647 --alph-size nucl:5,aa:21 --max-seq-len 65535 --max-seqs 20 --split 0 --split-mode 2 --split-memory-limit 0 -c 0.8 --cov-mode 0 --comp-bias-corr 0 --diag-score 0 --exact-kmer-matching 0 --mask 1 --mask-lower-case 0 --min-ungapped-score 0 --add-self-matches 0 --spaced-kmer-mode 1 --db-load-mode 0 --pca 1 --pcb 1.5 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Query database size: 67 type: Aminoacid\n",
      "Estimated memory consumption: 977M\n",
      "Target database size: 67 type: Aminoacid\n",
      "Index table k-mer threshold: 154 at k-mer size 6 \n",
      "Index table: counting k-mers\n",
      "[=================================================================] 100.00% 67 0s 7ms      \n",
      "Index table: Masked residues: 191\n",
      "Index table: fill\n",
      "[=================================================================] 100.00% 67 0s 1ms      \n",
      "Index statistics\n",
      "Entries:          4607\n",
      "DB size:          488 MB\n",
      "Avg k-mer size:   0.000072\n",
      "Top 10 k-mers\n",
      "    PGPRGS\t63\n",
      "    KPHFLI\t59\n",
      "    CRSGTT\t58\n",
      "    RFGLDK\t57\n",
      "    VQKDFT\t55\n",
      "    IGTNPG\t52\n",
      "    PGTTCR\t51\n",
      "    RAGPFS\t50\n",
      "    DEIFLW\t50\n",
      "    EAYCWY\t50\n",
      "Time for index table init: 0h 0m 0s 731ms\n",
      "Process prefiltering step 1 of 1\n",
      "\n",
      "k-mer similarity threshold: 154\n",
      "Starting prefiltering scores calculation (step 1 of 1)\n",
      "Query db start 1 to 67\n",
      "Target db start 1 to 67\n",
      "[=================================================================] 100.00% 67 0s 21ms     6 eta 0s       \n",
      "\n",
      "3.275860 k-mers per position\n",
      "1861 DB matches per sequence\n",
      "0 overflows\n",
      "0 queries produce too many hits (truncated result)\n",
      "20 sequences passed prefiltering per query sequence\n",
      "20 median result list length\n",
      "0 sequences with 0 size result lists\n",
      "Time for merging to pref_step0: 0h 0m 0s 2ms\n",
      "Time for processing: 0h 1m 37s 641ms\n",
      "align tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/pref_step0 tmp/5039645332348957457/clu_tmp/11355549737085215422/aln_step0 --sub-mat nucl:nucleotide.out,aa:blosum62.out -a 0 --alignment-mode 3 --alignment-output-mode 0 --wrapped-scoring 0 -e 0.001 --min-seq-id 0.9 --min-aln-len 0 --seq-id-mode 0 --alt-ali 0 -c 0.8 --cov-mode 0 --max-seq-len 65535 --comp-bias-corr 0 --max-rejected 2147483647 --max-accept 2147483647 --add-self-matches 0 --db-load-mode 0 --pca 1 --pcb 1.5 --score-bias 0 --realign 0 --realign-score-bias -0.2 --realign-max-seqs 2147483647 --gap-open nucl:5,aa:11 --gap-extend nucl:2,aa:1 --zdrop 40 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Compute score, coverage and sequence identity\n",
      "Query database size: 67 type: Aminoacid\n",
      "Target database size: 67 type: Aminoacid\n",
      "Calculation of alignments\n",
      "[=================================================================] 100.00% 67 0s 24ms     \n",
      "Time for merging to aln_step0: 0h 0m 0s 1ms\n",
      "635 alignments calculated\n",
      "314 sequence pairs passed the thresholds (0.494488 of overall calculated)\n",
      "4.686567 hits per query sequence\n",
      "Time for processing: 0h 0m 0s 33ms\n",
      "clust tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/aln_step0 tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_step0 --cluster-mode 0 --max-iterations 1000 --similarity-type 2 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering mode: Set Cover\n",
      "[=================================================================] 100.00% 67 0s 3ms      \n",
      "Sort entries\n",
      "Find missing connections\n",
      "Found 119 new connections.\n",
      "Reconstruct initial order\n",
      "[=================================================================] 100.00% 67 0s 0ms      \n",
      "Add missing connections\n",
      "[=================================================================] 100.00% 67 0s 0ms      \n",
      "\n",
      "Time for read in: 0h 0m 0s 23ms\n",
      "Total time: 0h 0m 0s 24ms\n",
      "\n",
      "Size of the sequence database: 67\n",
      "Size of the alignment database: 67\n",
      "Number of clusters: 33\n",
      "\n",
      "Writing results 0h 0m 0s 0ms\n",
      "Time for merging to clu_step0: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 39ms\n",
      "mergeclusters tmp/5039645332348957457/input tmp/5039645332348957457/clu tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_redundancy tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_step0 --threads 4 --compressed 0 -v 3 \n",
      "\n",
      "Clustering step 1\n",
      "[=================================================================] 100.00% 67 0s 5ms      \n",
      "Clustering step 2\n",
      "[=================================================================] 100.00% 33 0s 13ms     \n",
      "Write merged clustering\n",
      "[=================================================================] 100.00% 387 0s 15ms     \n",
      "Time for merging to clu: 0h 0m 0s 1ms\n",
      "Time for processing: 0h 0m 0s 28ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/input_step_redundancy_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 2ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/pref_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/aln_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_tmp/11355549737085215422/clu_step0 -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "createtsv tmp/5039645332348957457/input tmp/5039645332348957457/input tmp/5039645332348957457/clu tmp/5039645332348957457/cluster.tsv --threads 4 -v 3 \n",
      "\n",
      "Time for merging to cluster.tsv: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 42ms\n",
      "result2repseq tmp/5039645332348957457/input tmp/5039645332348957457/clu tmp/5039645332348957457/clu_rep --db-load-mode 0 --compressed 0 --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 33 0s 3ms      \n",
      "Time for merging to clu_rep: 0h 0m 0s 3ms\n",
      "Time for processing: 0h 0m 0s 45ms\n",
      "result2flat tmp/5039645332348957457/input tmp/5039645332348957457/input tmp/5039645332348957457/clu_rep tmp/5039645332348957457/rep_seq.fasta --use-fasta-header -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 1ms\n",
      "createseqfiledb tmp/5039645332348957457/input tmp/5039645332348957457/clu tmp/5039645332348957457/clu_seqs --threads 4 -v 3 \n",
      "\n",
      "[=================================================================] 100.00% 33 0s 1ms      \n",
      "Time for merging to clu_seqs: 0h 0m 0s 0ms\n",
      "Time for processing: 0h 0m 0s 27ms\n",
      "result2flat tmp/5039645332348957457/input tmp/5039645332348957457/input tmp/5039645332348957457/clu_seqs tmp/5039645332348957457/all_seqs.fasta -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/input -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/input_h -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_seqs -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu_rep -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "rmdb tmp/5039645332348957457/clu -v 3 \n",
      "\n",
      "Time for processing: 0h 0m 0s 0ms\n",
      "Mseq2 clustering complete. Writing data to /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/\n",
      "WARNING: Unsupported old command line option '-force' will be ignored (may change default output stream and format)\n",
      "WARNING: Your old-style command-line options were converted to:  ./clustalo -i /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta --dealign -o /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_4_rep_seq_aligned.fasta --outfmt=fasta -v --force\n",
      "Using 4 threads\n",
      "Read 34 sequences (type: Protein) from /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/mmseq_hmmer_data/step_3_temp_mmseq_rep_seq.fasta\n",
      "not more sequences (34) than cluster-size (100), turn off mBed\n",
      "Dealigning already aligned input sequences as requested.\n",
      "Calculating pairwise ktuple-distances...\n",
      "Ktuple-distance calculation progress done. CPU time: 0.11u 0.00s 00:00:00.11 Elapsed: 00:00:00\n",
      "Guide-tree computation done.\n",
      "Progressive alignment progress done. CPU time: 1.69u 0.09s 00:00:01.78 Elapsed: 00:00:01\n",
      "Alignment written to /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_4_rep_seq_aligned.fasta\n",
      "# hmmbuild :: profile HMM construction from multiple sequence alignments\n",
      "# HMMER 3.4 (Aug 2023); http://hmmer.org/\n",
      "# Copyright (C) 2023 Howard Hughes Medical Institute.\n",
      "# Freely distributed under the BSD open source license.\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "# input alignment file:             /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/step_4_rep_seq_aligned.fasta\n",
      "# output HMM file:                  /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/rep_seq_hmm\n",
      "# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -\n",
      "\n",
      "# idx name                  nseq  alen  mlen eff_nseq re/pos description\n",
      "#---- -------------------- ----- ----- ----- -------- ------ -----------\n",
      "1     step_4_rep_seq_aligned    34   450   219     0.95  0.591 \n",
      "\n",
      "# CPU time: 0.11u 0.00s 00:00:00.11 Elapsed: 00:00:00.11\n",
      "HMM-based alignment of fasta database is complete.\n"
     ]
    }
   ],
   "source": [
    "!python run_fido.py \\\n",
    "            -in /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/9b1_light_chain_top_500.fasta \\\n",
    "            -ref /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/9b1_light_chain.fasta \\\n",
    "            -out /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/top500/ \\\n",
    "            -add_ref /home/ubuntu/cookstore/250414_9b1/res/seq_data_light_chain/9b1_light_chain.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f36e62e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "brew",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
